core/dbt/graph/selector_methods.py

```diff
@@ -428,6 +428,10 @@ def recursively_check_macros_modified(self, node, previous_macros):
             macro_node = self.manifest.macros[macro_uid]
             if len(macro_node.depends_on.macros) > 0:
                 return self.recursively_check_macros_modified(macro_node, previous_macros)
+            # this macro hasn't been modified, but we haven't checked
+            # the other macros the node depends on, so keep looking
+            elif len(node.depends_on.macros) > len(previous_macros):
```

Comment:
This is nitpicking but I would recommend renaming `previous_macros` to something like `visited_macros`.